overlord/ifacestate/handlers.go

```diff
@@ -197,7 +198,19 @@ func (m *InterfaceManager) setupProfilesForSnap(task *state.Task, _ *tomb.Tomb,
 		affectedSnaps = append(affectedSnaps, name)
 	}
 	sort.Strings(affectedSnaps)
-	return m.setupAffectedSnaps(task, snapName, affectedSnaps)
+	err = m.setupAffectedSnaps(task, snapName, affectedSnaps)
+
+	// Store the snap revision in the state, so that we know which revision has profiles.
+	st := task.State()
+	var snapifst SnapInterfaceState
+	if err := Get(st, snapName, &snapifst); err != nil && err != state.ErrNoState {
```

Comment:
This `err` shadows the other `err`. So even if `Get()` fails, the function will return nil. Is it intentional?

Reply:
Are you sure? This definition should only apply to the if/then/else statement. It doesn't exist on line 213 below.

Reply:
To be clear to future reviewers: the error on Get is non-fatal and we deliberately don't want to return that.